pkg/kubelet/volume/populator/desired_state_of_world_populator.go

```diff
@@ -119,17 +136,67 @@ func (dswp *desiredStateOfWorldPopulator) findAndAddNewPods() {
 func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 	for _, volumeToMount := range dswp.desiredStateOfWorld.GetVolumesToMount() {
 		if _, podExists :=
-			dswp.podManager.GetPodByUID(volumeToMount.Pod.UID); !podExists {
-			glog.V(10).Infof(
-				"Removing volume %q (volSpec=%q) for pod %q from desired state.",
-				volumeToMount.VolumeName,
-				volumeToMount.VolumeSpec.Name(),
-				volumeToMount.PodName)
-
-			dswp.desiredStateOfWorld.DeletePodFromVolume(
-				volumeToMount.PodName, volumeToMount.VolumeName)
-			dswp.deleteProcessedPod(volumeToMount.PodName)
+			dswp.podManager.GetPodByUID(volumeToMount.Pod.UID); podExists {
+			continue
 		}
+
```

Comment:
This will still cause quite a lot of churn when deleting a large number of pods in a short time. The older code had less performance impact because it ran `docker ps` once and looked at all the volumes based on the output. 

E.g., https://github.com/kubernetes/kubernetes/blob/release-1.2/pkg/kubelet/kubelet.go#L2168